### PR TITLE
Add check for non-assignment code in rules.mk

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

As part of what we discussed in #8422 this adds a check for keyboard level `rules.mk` files to ensure that only assignment rules are present. We can change the name to whatever we decide on in.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by Erovia — APPROVED** [review]
Needs a rebase, but LGTM otherwise.

**Comment by tzarc** [comment]
So is this waiting for #8422 before merge? Preventing non-assignment options without a workable alternative implementation won't fly.

**Comment by skullydazed** [comment]
Yes, we actually need the post processing file before this can be merged.

**Review by tzarc — APPROVED** [review]

**Pull request merged**
